src/plugin.spec.ts

```diff
@@ -0,0 +1,204 @@
+import { PNG } from 'pngjs'
+import {
+  generateImage,
+  updateSnapshot,
+  compareSnapshots,
+  type CompareSnapshotOptions,
+  type UpdateSnapshotOptions
+} from './plugin'
+import { expect } from 'vitest'
+import { unlinkSync, existsSync } from 'node:fs'
+import path from 'node:path'
+
+function deleteFileSafely(filePath: string): void {
+  try {
+    unlinkSync(filePath)
+  } catch (err) {
+    if (err.code !== 'ENOENT') {
+      throw err // If the error is not "File not found" (ENOENT), rethrow it
+    } // If the error is "File not found," do nothing (fail silently)
+  }
+}
+
+const baseUpdateOptions: UpdateSnapshotOptions = {
+  screenshotName: 'enjuto',
+  specName: 'sub-folder',
+  screenshotAbsolutePath: path.join('fixtures', 'assets', 'base', 'enjuto.png'),
+  baseDirectory: path.join('fixtures', 'assets', 'base')
+} as const
+
+const baseCompareOptions: CompareSnapshotOptions = {
+  ...baseUpdateOptions,
+  errorThreshold: 0.1,
+  diffDirectory: path.join('cypress', 'snapshots', 'test', 'diff'),
+  generateDiff: 'fail'
+}
+
+const baseDirectoryCustom = path.join('output')
+const baseDirectoryDefault = path.join('cypress', 'snapshots', 'base')
+const diffFilePath = path.join(baseCompareOptions.diffDirectory ?? '', 'enjuto.png')
+const absolutePathMod = path.join('fixtures', 'assets', 'mod', 'enjuto.png')
+const validImagePath = path.join('mock', 'test.png')
+const systemFileName = path.join('/', 'System', 'assets', 'test.png')
+const wrongAbsolutePath = path.join('fixtures', 'assets', 'wadus.png')
+const rootFileName = path.join('/', 'test.png')
+const copiedFileName = path.join(
+  baseDirectoryDefault,
+  baseUpdateOptions.specName,
+  `${baseUpdateOptions.screenshotName}.png`
+)
+const copiedFileNameCustom = path.join(
+  baseDirectoryCustom,
+  baseUpdateOptions.specName,
+  `${baseUpdateOptions.screenshotName}.png`
+)
+
+const buildPNG = (): PNG => {
+  const png = new PNG({ width: 1, height: 1 })
+  png.data = Buffer.from([255, 0, 0, 255])
+  return png
+}
+
+describe('plugin', () => {
+  describe('generateImage', () => {
+    describe('when the image is generated', () => {
+      afterEach(() => {
+        deleteFileSafely(path.join(validImagePath))
```

Comment:
is there a reason to have a `path.join` with single argument?

Reply:
absolutely not ,  😅

Fixed thanks